controllers/discovery/searchUser.js

```diff
@@ -18,6 +18,11 @@ const SearchUser = async (req, res) => {
       message: 'Your search characters is too few, please input 3 or more characters for search'
     });
 
+  let where_anon_dm = '';
+  if (allow_anon_dm) {
+    where_anon_dm = 'AND u.allow_anon_dm = :allow_anon_dm';
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code block constructs a part of an SQL query based on the value of `allow_anon_dm`. This approach can lead to SQL injection if `allow_anon_dm` is not properly sanitized. Consider using parameterized queries or prepared statements instead.
```diff
- let where_anon_dm = '';
- if (allow_anon_dm) {
-   where_anon_dm = 'AND u.allow_anon_dm = :allow_anon_dm';
- }
+ let where_anon_dm = 'AND u.allow_anon_dm = :allow_anon_dm';
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->